### PR TITLE
Use BigDecimal for temperature conversion

### PR DESCRIPTION
**Pull request by cdjackson**
https://github.com/openhab/openhab/issues/3395

### CODE REVIEW

File: bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/converter/ZWaveMultiLevelSensorConverter.java [hunk 1]
```diff
@@ -106,16 +110,13 @@ public void handleEvent(ZWaveCommandClassValueEvent event, Item item, Map<String
...
-						val = new BigDecimal((c * 9.0 / 5.0) + 32.0 );
-					}
-					else if(sensorEvent.getSensorScale() == 1) {
-						// Scale is fahrenheit, convert to celsius
-						double f = ((BigDecimal)val).doubleValue();
-						val = new BigDecimal((f - 32.0) * 5.0 / 9.0 );					
-					}
+                    if (sensorEvent.getSensorScale() == 0) {
+                        // Scale is celsius, convert to fahrenheit
+                        val = val.multiply(ONE_POINT_EIGHT).add(THIRTY_TWO).movePointRight(1);
```
Comment (watou) [inline bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/converter/ZWaveMultiLevelSensorConverter.java]: The movePointRight and Left were for a vendor-specific F*10 to/from C implementation, so drop those! :)

File: bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/converter/ZWaveMultiLevelSensorConverter.java [hunk 2]
```diff
@@ -106,16 +110,13 @@ public void handleEvent(ZWaveCommandClassValueEvent event, Item item, Map<String
...
+                    } else if (sensorEvent.getSensorScale() == 1) {
+                        // Scale is fahrenheit, convert to celsius
+                        val = val.movePointLeft(1).subtract(THIRTY_TWO).divide(ONE_POINT_EIGHT, MathContext.DECIMAL32);
```
Comment (watou) [inline bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/converter/ZWaveMultiLevelSensorConverter.java]: ditto!

**Comment by cdjackson** [comment]
Thanks John :)

**Comment by watou** [comment]
62.60000000000000142108547152020037174224853515625 is now being reported as 62.6.  Thanks, Chris.

**Pull request merged**